app/assets/javascripts/dialog_field_refresh.js

```diff
@@ -135,6 +143,7 @@ var dialogFieldRefresh = {
       $('.dynamic-text-box-' + fieldId).val(responseData.values.text);
       dialogFieldRefresh.setReadOnly($('.dynamic-text-box-' + fieldId), responseData.values.read_only);
       dialogFieldRefresh.setVisible($('#field_' +fieldId + '_tr'), responseData.values.visible);
+      callback.call();
     };
 
```

Comment:
If easily doable, can you DRY up `refreshTextAreaBox` and `refreshTextBox`  since both functions are so identical?